packages/core/src/service/services/collection.service.ts

```diff
@@ -460,11 +465,19 @@ export class CollectionService implements OnModuleInit {
         }
         const preIdsSet = new Set(preIds);
         const postIdsSet = new Set(postIds);
-        const difference = [
-            ...preIds.filter(id => !postIdsSet.has(id)),
-            ...postIds.filter(id => !preIdsSet.has(id)),
-        ];
-        return difference;
+
+        if (applyToChangedVariantsOnly === undefined ? true : applyToChangedVariantsOnly) {
```

Comment:
This could be simplified to 
```ts
if (applyToChangedVariantsOnly ?? true) { }
```
where `??` is the nullish-coalescing operator, where the right-hand-side of the operator gets evaluated only if the left-hand side is "nullish" (null or undefined)

OR 

you could define a default value in the method signature, and then you can do a simplified if check: 
```ts
applyCollectionFiltersInternal(collection: Collection, applyToChangedVariantsOnly = true) {
   // ...
   if (applyToChangedVariantsOnly) {}
}
```

Reply:
Added second variant